app/models/post.rb

```diff
@@ -178,15 +178,24 @@ def last_seen_reply_for(user)
     @last_seen = reply
   end
 
-  def recent_characters_for(user, count)
-    # fetch the (count) most recent non-nil character_ids for user in post
-    recent_ids = replies.where(user_id: user.id)
-      .where.not(character_id: nil)
-      .limit(count)
-      .group('character_id')
-      .select('DISTINCT character_id, MAX(id)')
-      .order(Arel.sql('MAX(id) desc'))
-      .pluck(:character_id)
+  def recent_characters_for(user, count, multi_replies_params: nil)
+    # fetch the (count) most recent non-nil character_ids for user in post, including those being added by multi-replies
+    recent_ids = []
+    if multi_replies_params
+      recent_ids = multi_replies_params.reverse.pluck(:character_id).compact_blank.uniq.take(count).map(&:to_i)
+      count -= recent_ids.length
+    end
```

Comment:
You've otherwise been using an instance variable for `multi_replies_params`?

Reply:
This is on the post model though right? Previously it's on the controller

Reply:
Ah, so it is, and the method does pass it, so